pkg/util/azureclient/azuresdk/armauthorization/roledefinitions.go

```diff
@@ -17,13 +18,20 @@ type RoleDefinitionsClient interface {
 
 type ArmRoleDefinitionsClient struct {
 	*armauthorization.RoleDefinitionsClient
+	subscriptionID string
 }
 
 var _ RoleDefinitionsClient = &ArmRoleDefinitionsClient{}
 
-func NewArmRoleDefinitionsClient(credential azcore.TokenCredential, options *arm.ClientOptions) (*ArmRoleDefinitionsClient, error) {
+func NewArmRoleDefinitionsClient(credential azcore.TokenCredential, subscriptionID string, options *arm.ClientOptions) (*ArmRoleDefinitionsClient, error) {
 	client, err := armauthorization.NewRoleDefinitionsClient(credential, options)
 	return &ArmRoleDefinitionsClient{
 		RoleDefinitionsClient: client,
+		subscriptionID:        subscriptionID,
 	}, err
 }
+
+func (client ArmRoleDefinitionsClient) GetByID(ctx context.Context, roleDefinitionID string, options *armauthorization.RoleDefinitionsClientGetByIDOptions) (armauthorization.RoleDefinitionsClientGetByIDResponse, error) {
+	roleID := fmt.Sprintf("/subscriptions/%s%s", client.subscriptionID, roleDefinitionID)
```

Comment:
I recommend reworking this so that callers of this function only have to pass in the GUID portion of the role definition ID. Even if you were to explain it in a doc comment, I see potential for confusing errors if the caller doesn't understand that they have to pass `/providers/Microsoft.Authorization/roleDefinitions/xxx...` and that the leading slash at the front of the string is mandatory.

I had to trace through a few layers of function calls to understand what was going on here myself, but that wouldn't be necessary if the logic related to formatting the role definition ID was encapsulated within the role definition client.

Reply:
Updated the usage here to only take the roleDefinitionID, please confirm.